src/functionalTest/java/uk/gov/hmcts/reform/pip/account/management/controllers/BulkAccountTest.java

```diff
@@ -0,0 +1,115 @@
+
+import com.github.dockerjava.zerodep.shaded.org.apache.hc.core5.http.HttpHeaders;
+import io.restassured.response.Response;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.pip.account.management.utils.FunctionalTestBase;
+import uk.gov.hmcts.reform.pip.account.management.utils.OAuthClient;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@ExtendWith(SpringExtension.class)
+@ActiveProfiles(profiles = "functional")
+@SpringBootTest(classes = {OAuthClient.class})
+class BulkAccountTest extends FunctionalTestBase {
+    private static final String USER_ID = UUID.randomUUID().toString();
+    private static final String MOCK_FILE = "files/test-csv.csv";
+    private static final String BULK_UPLOAD_URL = "account/media-bulk-upload";
+    private static final String TESTING_SUPPORT_ACCOUNT_URL = "/testing-support/account/";
+    private static final String BEARER = "Bearer ";
+    private static final String ISSUER_ID = "x-issuer-id";
+
+    private Map<String, String> bearer;
+    private Map<String, String> issuerId;
+
+    @BeforeAll
+    void startUp() throws IOException {
+        bearer = Map.of(HttpHeaders.AUTHORIZATION, BEARER + accessToken);
+        issuerId = Map.of(ISSUER_ID, USER_ID);
+
+        StringBuilder csvContent = new StringBuilder(400);
+        csvContent.append("email,firstName,surname\n");
+
+        for (int i = 0; i < 5; i++) {
+            String email = generateTestEmail();
+            csvContent.append(email).append(",testBulkFirstName,testBulkSurname\n");
+        }
+
+        Path mockFilePath = Files.createTempFile("mock-bulk-upload", ".csv");
+        try (BufferedWriter writer = Files.newBufferedWriter(mockFilePath)) {
+            writer.write(csvContent.toString());
+        }
+
+        System.setProperty(MOCK_FILE, mockFilePath.toString());
+    }
+
+    private String generateTestEmail() {
+        String prefix = String.format("pip-am-test-email-%s", ThreadLocalRandom.current().nextInt(1000, 9999));
+        return prefix + "@justice.gov.uk";
+    }
+
+    @AfterAll
+    public void teardown() throws IOException {
+        File mockFile = new File(System.getProperty(MOCK_FILE));
+
+        List<String> emails = Files.lines(mockFile.toPath(), StandardCharsets.UTF_8)
```

Comment:
Rather than needing to parse the original file, I'm thinking you could use a standard prefix that all accounts in this test use.

E.g pip-am-test-email-<Randomly generated ID for Test>-<Randomly Generated ID for Account>

This way you'll be able to call the delete request just once with pip-am-test-email-<Randomly generated ID for Test> and it'll delete them all